ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/keys/ContractKeysValidation.scala

```diff
@@ -157,4 +177,8 @@ private[transaction] object ContractKeysValidation {
 
   private[keys] type KeyValidationStatus =
     Either[KeyValidationError, KeyValidationState]
+
+  def onCurrentStatus[E, A](
```

Comment:
This is what I mean by staying close to `main` rather than the previous state. That said, happy to drop it if you prefer.